### PR TITLE
Add a support for PHAR building

### LINKED ISSUE #54: Release a standalone version (.phar) with external config file
In order to use JoliTypo in other contexts, it could be useful to release a .phar version, which can be configured with a .jolitypo configuration file.

It could be used with all static site generators, after build phase to fix the generated HTML files.

It looks [Box](https://github.com/box-project/box) could help use to generate the .phar

### LINKED ISSUE #62: Distribute a PHAR version of JoliTypo
It could be useful to make available a standalone binary of JoliTypo, to make it usable outside of the PHP ecosystem (for all JS/Go static site generators for example).

May be [Box](https://github.com/box-project/box) may help us to craft the PHAR version.

### PR DESCRIPTION
**Pull request by lyrixx**
fix #54
fix #63
fix #62

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +55,13 @@ $fixer = new Fixer(array("Trademark", "SmartQuotes"));
 $fixedContent = $fixer->fixString('Here is a "protip(c)"!'); // Here is a “protip©”!
 ```
 
+CLI usage
+=========
+
+You can run a standalone version of JoliTypo by downloading [the PHAR version](https://github.com/jolicode/JoliTypo/releases/latest)
+
+Run `jolitypo --help` to know how to configure the Fixer.
```
Comment (welcoMattic) [inline README.md:63]:
```suggestion
Run `jolitypo.phar --help` to know how to configure the Fixer.
```

File: tools/phar/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# JoliTypo PHAR building
+
+## Usage
+
+```bash
+composer install
+vendor/bin/box  compile
```
Comment (welcoMattic) [inline tools/phar/README.md:7]:
```suggestion
vendor/bin/box compile
```

File: tools/phar/composer.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        "symfony/console": "^5.3",
+        "symfony/finder": "^5.3"
+    },
+    "require-dev": {
+        "humbug/box": "^3.13",
+        "humbug/php-scoper": "dev-master#d3c7b5cde60ce6e269a202d96ba0549ac7528319 as 0.14.0"
+    },
+    "autoload": {
+        "psr-4": {
+            "Jolicode\\JolitypoPhar\\": "src/"
```
Comment (welcoMattic) [inline tools/phar/composer.json:22]:
```suggestion
            "JoliCode\\JolitypoPhar\\": "src/"
```

**Review by welcoMattic — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (lyrixx) [reply README.md:63]: Nope. the suffix is useless to me 
(Like composer)

**Comment by welcoMattic** [comment]
https://github.com/jolicode/JoliTypo/blob/039246f9e2b16c530bfc99056d8688561b2592f7/tools/phar/a.html should be removed, no?

File: tools/phar/composer.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
+    "name": "jolicode/jolitypo-phar",
+    "description": "Tooling for building JoliTypo PHAR",
+    "type": "project",
+    "license": "MIT",
+    "authors": [
+        {
+            "name": "Grégoire Pineau",
+            "email": "lyrixx@lyrixx.info"
+        }
```
Comment (welcoMattic) [inline tools/phar/composer.json:10]:
```suggestion
        },
        {
            "name": "Mathieu Santostefano",
            "email": "msantostefano@pm.me"
        }
```

**Comment by lyrixx** [comment]
@welcoMattic I addressed your comment 👍🏼

**Review by damienalexandre — APPROVED** [review]
A note in the Changelog could be added too. 

I'm not a big fan of the hardcoded `$recommendedRulesByLocale` also. That should either be in the main library, or nowhere IMO.

**Comment by lyrixx** [comment]
> I'm not a big fan of the hardcoded `$recommendedRulesByLocale` also. That should either be in the main library, or nowhere IMO.

I moved it to the main lib, and I also added the phar compilation in the CI

**Review by damienalexandre — APPROVED** [review]

**Pull request merged**